blockchain/blockchain.go

```diff
@@ -874,6 +874,68 @@ func (b *Blockchain) executeBlockTransactions(block *types.Block) (*BlockResult,
 	}, nil
 }
 
+// WriteFullBlock writes a single block to the local blockchain.
+// It doesn't do any kind of verification, only commits the block to the DB
+// This function is a copy of WriteBlock but with a full block which does not
+// require to compute again the Receipts.
+func (b *Blockchain) WriteFullBlock(fblock *types.FullBlock, source string) error {
```

Comment:
I wrote this function instead of changing the other one (`WriteBlock`) in order to avoid any issues with `ibft` consensus protocol.